### PR TITLE
[WIP] Glue plugin to work with Glue 0.11

### LINKED ISSUE #22: Glue plugin in Ginga broken with Glue 0.11
I am using dev version of Ginga, Glue 0.11 from `pip`, and dev verson of `glue_ginga`. When I  press "Start Glue" in the plugin, I get:
```
Error making callback 'activated': __init__() got an unexpected keyword argument 'maximized'
Traceback:
  File ".../ginga/misc/Callback.py", line 128, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../glue_ginga/plugins/Glue.py", line 164, in <lambda>
    b.start_glue.add_callback('activated', lambda w: self.start_glue_cb())

  File ".../glue_ginga/plugins/Glue.py", line 305, in start_glue_cb
    self.glue_app = qglue()

  File ".../glue_ginga/plugins/Glue.py", line 378, in qglue
    ga = GlueApplication(data_collection=dc, maximized=False)
```

I have a vague recollection of adding the `maximized` keyword myself and this used to work, though I have not used it in a while.

c/c @astrofrog 

xref https://github.com/ejeschke/glue-ginga/pull/20#issuecomment-324969576

### PR DESCRIPTION
**Pull request by pllim**
Fix #22 

The first commit is as suggested in https://github.com/ejeschke/glue-ginga/issues/22#issuecomment-325045547 . However this does not fix other problems.

When I use the Glue plugin with the fix from the first commit, I see some cryptic message:
```
QTextCursor::setPosition: Position '-1' out of range
QTextCursor::setPosition: Position '10' out of range
```

Then, Ginga refuses to load image properly after Glue plugin is started.

Also, when I click on the "x" to close Glue started from within this plugin, I get a bunch of traceback related to Qt not able to find stuff, which is sometimes followed by a core dump.

tl;dr -- More investigation is needed to make this plugin work again under Ginga dev and Glue 0.11.

### CODE REVIEW

**Comment by astrofrog** [comment]
The setPosition warnings are due to a bug in ``qtconsole`` which is fixed in qtconsole dev (doesn't impact glue though) - I'm not sure what the other issues are though (away for a few days so can't work on this at the moment)

**Comment by pllim** [comment]
@ejeschke , perhaps this is also due to some refactoring you have done to Ginga internals in the past few months? Please advise.

**Comment by ejeschke** [comment]
@pllim, it is working for me, so long as I just remove the maximize=True, but leave `show()` as `show()` as I commented in #22.

**Comment by ejeschke** [comment]
Note that I just tested the basics: loading and transferring data back and forth. There may be other issues due to the changes in glue, but so far it looks ok to me.

**Comment by pllim** [comment]
Appveyor + Python 2 + Glue dev failed with this error (despite the actual tests passing):
```
cmd : INFO menuinst_win32:__init__(182): Menu: name: 'Anaconda${PY_VER} 
${PLATFORM}', prefix: 'C:\conda\envs\test', env_name: 'test', mode: 'None', 
used_mode: 'system'
At C:\projects\glue-ginga\ci-helpers\appveyor\install-miniconda.ps1:241 char:13
+   $output = cmd /c conda install -n test $QUIET $NUMPY_OPTION $CONDA_ ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (INFO menuinst_w..._mode: 'system' 
   :String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
INFO menuinst_win32:__init__(182): Menu: name: 'Anaconda${PY_VER} 
${PLATFORM}', prefix: 'C:\conda\envs\test', env_name: 'test', mode: 'None', 
used_mode: 'system'
```

File: glue_ginga/plugins/Glue.py [hunk 1]
```diff
@@ -309,8 +309,8 @@ def start_glue_cb(self):
 
         # self.glue_app._create_terminal()
         sgeo = self.glue_app.app.desktop().screenGeometry()
-        self.glue_app.resize(sgeo.width() * 0.9, sgeo.height() * 0.9)
         self.glue_app.show()
```
Comment (pllim) [inline glue_ginga/plugins/Glue.py:312]: NOTE: Using `show()` here instead of `start()` caused IPython terminal to be unavailable in Glue launch from the plugin. See https://github.com/ejeschke/glue-ginga/issues/22#issuecomment-325498218

**Comment by astrofrog** [comment]
I've seen the appveyor issue before, it's I think to do with downloads timing out. Restarting usually fixes it.

File: glue_ginga/plugins/Glue.py (hunk 1 above)
Comment (astrofrog) [reply glue_ginga/plugins/Glue.py:312]: And just to be clear, what happens if you do ``start``?

Comment (pllim) [reply glue_ginga/plugins/Glue.py:312]: After Glue is started from within the plugin, Ginga cannot load new images properly anymore. Sometimes, the buffer loads on main display (in Ginga) but thumbnail does not show, sometimes nothing loads at all. Basically, Ginga stops working.

**Comment by pllim** [comment]
> I'd recommend at least just merging the fix for the 'maximize' problem.

@ejeschke , you can certainly merge this now if you want knowing that IPython terminal won't work yet. @astrofrog can always file a follow-up PR for that part.

File: glue_ginga/plugins/Glue.py (hunk 1 above)
Comment (ejeschke) [reply glue_ginga/plugins/Glue.py:312]: > Basically, Ginga stops working.

We need to start glue without committing to it's Qt event loop.  The Ginga event loop will make calls back into glue to handle any QT events.

**Comment by ejeschke** [comment]
Perhaps the problem is a minor one that could be corrected by simply calling some initialization code in glue before or after show().  If so, it would be worth waiting a little to see if @astrofrog can fix it before bumping the release.

**Comment by pllim** [comment]
As advised by @astrofrog , I am merging this now so he can put the IPython fix on top of this. Thank you!

**Comment by astrofrog** [comment]
I'm looking into this - how do you start glue from ginga? I don't see glue in the list of ginga plugins.

**Comment by pllim** [comment]
```
ginga --modules=glue\_ginga.plugins.Glue
```

**Comment by pllim** [comment]
You can also throw in `--loglevel=40 --stderr` to print error to screen.

**Comment by ejeschke** [comment]
You can also just:
```bash
 mkdir $HOME/.ginga/plugins
cp Glue.py $HOME/.ginga/plugins
```
Once it is working well again we could transition it to the new external plugin template.
